ground/src/main/java/com/google/android/ground/MainViewModel.kt

```diff
@@ -135,14 +136,21 @@ constructor(
       .saveUser(user)
       .andThen(
         if (termsOfServiceRepository.isTermsOfServiceAccepted) {
-          Observable.just(HomeScreenFragmentDirections.showHomeScreen())
+          Observable.just(getNavDirectionForSignedInUser())
         } else {
           termsOfServiceRepository.termsOfService
             .map { SignInFragmentDirections.showTermsOfService().setTermsOfServiceText(it.text) }
             .cast(NavDirections::class.java)
-            .switchIfEmpty(Maybe.just(HomeScreenFragmentDirections.showHomeScreen()))
+            .switchIfEmpty(Maybe.just(getNavDirectionForSignedInUser()))
             .toObservable()
         }
       )
   }
+
+  private fun getNavDirectionForSignedInUser(): NavDirections =
```

Comment:
What does this method have to do with which user is signed in? Looks like it's forked based on whether there was a survey previously active.. so maybe this is `getDirectionsAfterSignIn`?